mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithMultipleCatalogs.java

```diff
@@ -0,0 +1,143 @@
+  @Parameterized.Parameter(5)
+  public TestTables.TestTableType testTableType2;
+  @Parameterized.Parameter(6)
+  public String table2CatalogName;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  private TestTables testTables1;
+  private TestTables testTables2;
+
+  @Parameterized.Parameters(name = "fileFormat1={0}, fileFormat2={1}, engine={2}, tableType1={3}, catalogName1={4}, " +
+          "tableType2={5}, catalogName2={6}")
+  public static Collection<Object[]> parameters() {
+    Collection<Object[]> testParams = new ArrayList<>();
+    String javaVersion = System.getProperty("java.specification.version");
+
+    // Run tests with PARQUET and ORC file formats for a two Catalogs
+    for (String engine : EXECUTION_ENGINES) {
+      // include Tez tests only for Java 8
+      if (javaVersion.equals("1.8") || "mr".equals(engine)) {
+        for (TestTables.TestTableType testTableType : TestTables.ALL_TABLE_TYPES) {
+          if (!TestTables.TestTableType.HIVE_CATALOG.equals(testTableType)) {
+            testParams.add(new Object[]{FileFormat.PARQUET, FileFormat.ORC, engine,
+                TestTables.TestTableType.HIVE_CATALOG, HIVECATALOGNAME, testTableType, OTHERCATALOGNAME});
+          }
+        }
+      }
+    }
+    return testParams;
+  }
+
+  @BeforeClass
+  public static void beforeClass() {
+    shell = HiveIcebergStorageHandlerTestUtils.shell();
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    shell.stop();
+  }
+
+  @Before
+  public void before() throws IOException {
+    testTables1 = HiveIcebergStorageHandlerTestUtils.testTables(shell, testTableType1, temp, table1CatalogName);
+    HiveIcebergStorageHandlerTestUtils.init(shell, testTables1, temp, executionEngine);
+    testTables1.properties().entrySet().forEach(e -> shell.setHiveSessionValue(e.getKey(), e.getValue()));
+
+    testTables2 = HiveIcebergStorageHandlerTestUtils.testTables(shell, testTableType2, temp, table2CatalogName);
+    testTables2.properties().entrySet().forEach(e -> shell.setHiveSessionValue(e.getKey(), e.getValue()));
+  }
+
+  @After
+  public void after() throws Exception {
+    HiveIcebergStorageHandlerTestUtils.close(shell);
+  }
+
+  @Test
+  public void testJoinTablesFromDifferentCatalogs() throws IOException {
+    createAndAddRecords(testTables1, fileFormat1, TableIdentifier.of("default", "customers1"), table1CatalogName,
+        HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS);
+    createAndAddRecords(testTables2, fileFormat2, TableIdentifier.of("default", "customers2"), table2CatalogName,
+        HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS);
+
+    List<Object[]> rows = shell.executeStatement("SELECT c2.customer_id, c2.first_name, c2.last_name " +
+            "FROM default.customers2 c2 JOIN default.customers1 c1 ON c2.customer_id = c1.customer_id " +
+            "ORDER BY c2.customer_id");
+    Assert.assertEquals(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS.size(), rows.size());
```

Comment:
nit, but the size check already takes place within `HiveIcebergTestUtils.validateData`